src/main/scala/com/databricks/labs/automl/sanitize/FeatureCorrelationDetection.scala

```diff
@@ -54,30 +54,34 @@ class FeatureCorrelationDetection(data: DataFrame, fieldListing: Array[String])
   private def computeFeatureCorrelation(): Array[FeatureCorrelationStats] = {
 
     val correlationInteractions = new ArrayBuffer[FeatureCorrelationStats]
-    val redundantRecursionEliminator = new ArrayBuffer[String]
-
+   
     val taskSupport = new ForkJoinTaskSupport(new ForkJoinPool(_parallelism))
     val fieldListingPar = fieldListing.par
     fieldListingPar.tasksupport = taskSupport
 
-
-    fieldListingPar.foreach{ x =>
-      val leftFields = fieldListing.filterNot(_.contains(x)).filterNot(f => redundantRecursionEliminator.contains(f))
-      leftFields.foreach{y =>
+   //In the previous code, both of the features having higher negative or positive correlation were getting removed
+   //because for every pair of feature a and b, in correlationInteraction (a,b) as well as (b,a) was getting added
+    fieldListingPar.foreach(cfeature =>
+    {
+      var currIndex = fieldListingPar.indexOf(cfeature) //get current index
+      while (currIndex +1 < fieldListingPar.length)    //for every index iterate over next indices only, so that for every (a,b) there won't be (b,a)
+      {
+        val nextIndex = currIndex + 1
+        val nfeature = fieldListing(nextIndex)      
         val corrStats: Double = try {
-          data.groupBy().agg(corr(x, y).as("pearson")).first().getDouble(0)
-        } catch {
+          dataFrame.groupBy().agg(corr(cfeature, nfeature).as("pearson")).first().getDouble(0)
+        }
+        catch {
           case e: java.lang.NullPointerException =>
-            val errorMsg = s"Correlation Calculation for $x : $y failed.  Recording Inf for correlation."
-            println(errorMsg)
+            val errorMsg = s"Correlation Calculation for $cfeature : $nfeature failed.  Recording Inf for correlation."
+
             logger.log(Level.INFO, errorMsg + s"\n ${e.printStackTrace()}")
             Double.PositiveInfinity
         }
-        correlationInteractions += FeatureCorrelationStats(x, y, corrStats)
+        currIndex+=1
```

Comment:
Hi @BhushG , a unit test to confirm and document the behavior would be super helpful.